src/dev/i18n/extract_default_translations.js

```diff
@@ -73,51 +115,83 @@ export async function extractDefaultTranslations(inputPath) {
       [hbsEntries, extractHandlebarsMessages],
     ].map(async ([entries, extractFunction]) => {
       const files = await Promise.all(
-        entries.map(async entry => {
-          return {
-            name: entry,
-            content: await readFileAsync(entry),
-          };
-        })
+        entries
+          .filter(entry => !exclude.includes(normalize(path.relative('.', entry))))
+          .map(async entry => {
+            return {
+              name: entry,
+              content: await readFileAsync(entry),
+            };
+          })
       );
 
       for (const { name, content } of files) {
         try {
           for (const [id, value] of extractFunction(content)) {
-            addMessageToMap(defaultMessagesMap, id, value);
+            validateMessageNamespace(id, name);
+            addMessageToMap(targetMap, id, value);
           }
         } catch (error) {
           throw new Error(`Error in ${name}\n${error.message || error}`);
         }
       }
     })
   );
+}
 
-  // .slice(0, -1): remove closing curly brace from json to append messages
-  let jsonBuffer = Buffer.from(
-    JSON5.stringify({ formats: i18n.formats }, { quote: `'`, space: 2 }).slice(0, -1)
-  );
+export async function extractDefaultTranslations({ paths, output, outputFormat }) {
+  const defaultMessagesMap = new Map();
+
+  for (const inputPath of filterPaths(paths)) {
+    await extractMessagesFromPathToMap(inputPath, defaultMessagesMap);
+  }
+
+  // messages shouldn't be extracted to a file if output is not supplied
+  if (!output || !defaultMessagesMap.size) {
+    return;
+  }
 
   const defaultMessages = [...defaultMessagesMap].sort(([key1], [key2]) => {
     return key1 < key2 ? -1 : 1;
   });
 
-  for (const [mapKey, mapValue] of defaultMessages) {
-    jsonBuffer = Buffer.concat([
-      jsonBuffer,
-      Buffer.from(`  '${mapKey}': '${mapValue.message}',`),
-      Buffer.from(mapValue.context ? ` // ${mapValue.context}\n` : '\n'),
-    ]);
-  }
+  const defaultMessagesObject = { formats: i18n.formats };
 
-  // append previously removed closing curly brace
-  jsonBuffer = Buffer.concat([jsonBuffer, Buffer.from('}\n')]);
+  if (outputFormat === 'json5') {
```

Comment:
__nit:__ it's a bit hard to read, would you mind moving `json` and `json5` serialization logic into two separate functions and then do something like this?
```js
await writeFileAsync(
    path.resolve(output, 'en.json'),
    outputFormat === 'json5'
      ? serializeToJson5(defaultMessages)
      : serializeToJson(defaultMessages)
  );
```